pandas/core/frame.py

```diff
@@ -2232,8 +2232,7 @@ def maybe_reorder(
 
             result_index = None
             if len(arrays) == 0 and index is None and length == 0:
-                # for backward compat use an object Index instead of RangeIndex
```

Comment:
I don't really understand this comment, looks like it was added in https://github.com/pandas-dev/pandas/pull/43104/files by @jbrockmendel  - any chance you could clarify? Is it still necessary?

Reply:
IIRC this was the existing behavior at the time and was added just to avoid changing it